pipe.go

```diff
@@ -223,9 +224,36 @@ func DialPipe(path string, timeout *time.Duration) (net.Conn, error) {
 // DialPipeContext attempts to connect to a named pipe by `path` until `ctx`
 // cancellation or timeout.
 func DialPipeContext(ctx context.Context, path string) (net.Conn, error) {
+	conn, err := DialPipeContextAccess(ctx, path, syscall.GENERIC_READ|syscall.GENERIC_WRITE)
+	if err == context.DeadlineExceeded {
+		return nil, ErrTimeout
+	}
+	return conn, err
+}
+
+// DialPipeContext attempts to connect to a named pipe by `path` until `ctx`
+// cancellation or timeout.
+func DialPipeWithAccess(path string, timeout *time.Duration, access uint32) (net.Conn, error) {
```

Comment:
We dont need two overloads of this. The non `context` version is legacy. Just add `DialPipeAccess(ctx context.Context, path string, access uint32)`